doc/calling-javascript.md

```diff
@@ -176,6 +176,26 @@ or as
 js.Dynamic.literal("foo" -> 42, "bar" -> "foobar")
 {% endhighlight %}
 
+### Literal object construction using an Scala object interface
+Sometimes for a nicer interface, Literal objects can be implimented using 
+a trait interface. 
+The above JavaScript code can be implemented using following code:
+
+{% highlight scala %}
+trait MyObject extends js.Object {
+  val foo: Int = js.native
+  val bar: String = js.native
+}
+{% endhighlight %}
+An scala object can be added for easy access:
```

Comment:
Also maybe say "should" instead of "can". And replace "easy access" (we don't really access, do we) by "type safe creation". (this is why one should do this, rather than scattering `js.Dynamic.literal` all over the code).